File: services/api/internal/model/wishlist.go
package model

import "time"

type WishlistItem struct {
	ID         string    `json:"id"`
	UserID     string    `json:"user_id"`
	TmdbID     int       `json:"tmdb_id"`
	Title      string    `json:"title"`
	Year       int       `json:"year"`
	PosterPath string    `json:"poster_path"`
	AddedAt    time.Time `json:"added_at"`
}

type AddWishlistRequest struct {
	TmdbID     int    `json:"tmdb_id" binding:"required"`
	Title      string `json:"title" binding:"required"`
	Year       int    `json:"year"`
	PosterPath string `json:"poster_path"`
}

type WishlistResponse struct {
	Items []WishlistItem `json:"items"`
	Total int            `json:"total"`
}

File: apps/moovi/src/lib/api/wishlist.ts
import { request } from './client'

export interface WishlistItem {
  id: string
  user_id: string
  tmdb_id: number
  title: string
  year: number
  poster_path: string
  added_at: string
}

export interface WishlistResponse {
  items: WishlistItem[]
  total: number
}

export interface AddWishlistPayload {
  tmdb_id: number
  title: string
  year: number
  poster_path: string
}

export async function getWishlist(): Promise<WishlistResponse> {
  return request<WishlistResponse>('/api/wishlist')
}

export async function addToWishlist(payload: AddWishlistPayload): Promise<WishlistItem> {
  return request<WishlistItem>('/api/wishlist', {
    method: 'POST',
    body: JSON.stringify(payload),
  })
}

export async function removeFromWishlist(tmdbId: number): Promise<void> {
  await request(`/api/wishlist/${tmdbId}`, {
    method: 'DELETE',
  })
}
